pkgs/build-support/coq/extra-lib.nix

```diff
@@ -1,22 +1,41 @@
 { lib }:
-with builtins; with lib; recursiveUpdate lib (rec {
 
-  versions =
-    let
-      truncate = n: v: concatStringsSep "." (take n (splitVersion v));
-      opTruncate = op: v0: v: let n = length (splitVersion v0); in
-         op (truncate n v) (truncate n v0);
-    in rec {
+let
+  inherit (lib)
+    all
+    concatStringsSep
+    findFirst
+    flip
+    getAttr
+    head
+    isFunction
+    length
+    recursiveUpdate
+    splitVersion
+    tail
+    take
+    versionAtLeast
+    versionOlder
+    zipListsWith
+    ;
+
+  /* Get string of the first n parts of a version string.
 
-    /* Get string of the first n parts of a version string.
+     Example:
+     - truncate 2 "1.2.3-stuff"
+       => "1.2"
 
-       Example:
-       - truncate 2 "1.2.3-stuff"
-         => "1.2"
+     - truncate 4 "1.2.3-stuff"
+       => "1.2.3.stuff"
+  */
+  truncate = n: v: concatStringsSep "." (take n (splitVersion v));
 
-       - truncate 4 "1.2.3-stuff"
-         => "1.2.3.stuff"
-    */
+  opTruncate = op: v0: v: let n = length (splitVersion v0); in
+    op (truncate n v) (truncate n v0);
+
+  range = low: high: pred.inter (versions.isGe low) (versions.isLe high);
```

Comment:
I don't think this is necessary, because you're pulling the definition out of it's natural context.

Reply:
Reverted https://github.com/NixOS/nixpkgs/pull/295946/commits/8b84f3ace020939bd4a88df8e5b56a77180c22aa.